src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/LambdaRewriter.cs

```diff
@@ -1120,7 +1119,9 @@ public override BoundNode VisitConversion(BoundConversion conversion)
                 if (conversion.ConversionKind == ConversionKind.MethodGroup &&
                     conversion.SymbolOpt?.MethodKind == MethodKind.LocalFunction)
                 {
-                    return conversion;
+                    return new PartiallyLoweredLocalFunctionReference(
+                        conversion,
+                        new Dictionary<Symbol, CapturedSymbolReplacement>(proxies));
```

Comment:
The fact that `proxies` needs to be cloned seems like a possible source of bugs. To reduce the risk, consider extracting a helper method for creating the `PartiallyLoweredLocalFunctionReference` from the underlying node and cloning `proxies`, to use across the several places where references are created.